main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+var testData = []byte(`
+	{"Records": [
+		{"foo": "bar"},
+		{"key": {"subkey": "value"}}
+	]}
+`)
```

Comment:
might want to test incomplete and unclosed records, no records key, or just throw go-fuzz at this

Reply:
Those will all definitely error out, which atm I believe is ok. There is a well defined structure to Cloudtrail logs and I imagine if another file found it's way into these buckets we'd want it to error out? Am I thinking about this correctly?

Reply:
Yeah, I guess since this isn't a long running process exiting is fine.